base_agent/memory_nodes.py

```diff
@@ -789,17 +807,92 @@ def create(cls, memory, task: Task) -> str:
             >>> create(memory, task)
         """
+        old_memid = getattr(task, "memid", None)
+        if old_memid:
+            return old_memid
         memid = cls.new(memory)
         task.memid = memid  # FIXME: this shouldn't be necessary, merge Task and TaskNode?
-        memory.db_write(
-            "INSERT INTO Tasks (uuid, action_name, pickled, created_at) VALUES (?,?,?,?)",
+        memory._db_write(
+            "INSERT INTO Tasks (uuid, action_name, pickled, prio, running, created_at) VALUES (?,?,?,?,?,?)",
             memid,
             task.__class__.__name__,
             memory.safe_pickle(task),
+            task.prio,
+            task.running,
             memory.get_time(),
         )
         return memid
 
+    def step(self, agent):
+        self.task.step(agent)
+        self.update_task()
+
+    def update_task(self, task=None):
+        task = task or self.task
+        self.memory.db_write(
+            "UPDATE Tasks SET pickled=? WHERE uuid=?", self.memory.safe_pickle(task), self.memid
+        )
+
+    # FIXME TODO don't need paused, set prio to 0 and have a condtion for unpausing
+    # use this to update prio or running, don't do it directly on task or in db!!
+    def get_update_status(self, status, force_db_update=True, force_task_update=True):
+        """
+        status is a dict with possible keys "prio", "running", "finished".
+        
+        prio > 0 :  run me if possible, check my stop condition
+        prio = 0 :  check my start condition, run if true
+        prio < 0 :  don't even check my start condition
+
+        running = 1 :  task should be stepped if possible and not explicitly paused
+        running = 0 :  task should not be stepped
+
+        finished >  0 :  task has run and is complete, completed at the time indicated
+        finished = -1 :  task has not completed
+        
+        this method updates these columns of the DB for each of the keys if have values
+        if force_db_update is set, these will be updated with the relevant attr from self.task
+        even if it is not in the status dict.  
+        if force_task_update is set, the information will go the other way,
+        and whatever is in the dict will be put on the task.
+        """
+        status_out = {}
+        for k in ["prio", "running", "finished"]:
+            # update the task itself, hopefully don't need to do this when task objects are re-written as MemoryNode s
+            if force_task_update:
+                setattr(self.task, k, status.get(k) or getattr(self.task, k))
+            status_out[k] = getattr(self.task, k)
+            if k == "finished":
+                if self.task.finished:
+                    status_out[k] = self.agent_memory.get_time()
+                else:
+                    status_out[k] = -1
+
+            if status.get(k) or force_db_update:
+                key = k
+                if k == "finished":
+                    key = "finished_at"
```

Comment:
we can just save this info in `status_out ` with the key `finished_at` to avoid confusion here

Reply:
then the input status dict would be like "finished_at": True instead of "finished": True, that's confusing to me?  We could changed the db column to be "finished" instead of "finished_at" more generally

Reply:
Sure the latter sounds good.